### PR TITLE
chore: upgrade tiledb to 0.21.4

### LINKED ISSUE #4893: Update tiledb python library to latest version
As of this writing, the latest `tiledb` python library is `0.21.4` : https://pypi.org/project/tiledb/#history

This needs to be fixed because the `requirements.txt` [has pinned `tiledb` at 0.16.5](https://github.com/chanzuckerberg/single-cell-data-portal/blob/4b4fb4384f414e271c828255c05ca8b4b2b55895/requirements.txt#LL25C12-L25C12) which is starting to really lag the latest release and given that we depend on this database, we shouldn't wait too long to upgrade as that will exact a lot of pain when we are forced to upgrade.

Moreover, **cannot be installed on M1 Macs**. Only newer versions of tiledb can be `pip install` on M1 Macs. This means that docker containers (which runs version `0.16.5`) is different from `0.21.*` that one can install in their M1 Mac python virtual environment.

This means running _unit tests locally without docker_ fails due to backward incompatibility in the `tiledb` python API between `0.16.5` and `0.21.*`.

One example in API incompatibility is this:

#### 0.16.5

Calling `query` method is done this done by passing in `QueryCondition` object

`query_result_df = cube.query(attr_cond=attr_cond, use_arrow=True).df[tiledb_dims_query]` (https://github.com/chanzuckerberg/single-cell-data-portal/blob/4b4fb4384f414e271c828255c05ca8b4b2b55895/backend/wmg/data/query.py#L157)

#### 0.21.3

Here the `QueryCondition` object has been discarded and a string version of the query expression is passed in if the string is non-empty or `None` is passed in if the string is empty.

```
query_result_df = cube.query(cond=(query_cond_str if query_cond_str else None), use_arrow=True).df[tiledb_dims_query]
```

Best to upgrade when the API changes needed on the call sites are not too difficult to do.

### PR DESCRIPTION
**Pull request by ebezzi**
## Reason for Change

- Fixes #4893 

## Changes

- Upgrades tiledb to 0.21.1, which is the version pinned in Explorer.

## Testing steps

- Run tests in GHA, see if they pass
- Run this in rdev, upload a dataset, see if it can be opened by the Explorer
- For WMG, need to figure out a testing strategy w/ data-vis team

### CODE REVIEW

**Comment by prathapsridharan** [comment]
@ebezzi @atarashansky

Currently, this would only verify that the reader can read data written by the `0.16.5` version of `tiledb`. We would also need to test:

1. `tiledb 0.21.1` can be used by the WMG pipeline to generate cubes without problems
2. The WMG API can read the data generated in (1) above

Can we trigger the WMG pipeline run for this as well to ensure that the pipeline generates cubes without problems and can be read by the WMG API? Perhaps in staging environment?

**Comment by ebezzi** [comment]
> @ebezzi @atarashansky
> 
> Currently, this would only verify that the reader can read data written by the `0.16.5` version of `tiledb`. We would also need to test:
> 
> 1. `tiledb 0.21.1` can be used by the WMG pipeline to generate cubes without problems
> 2. The WMG API can read the data generated in (1) above
> 
> Can we trigger the WMG pipeline run for this as well to ensure that the pipeline generates cubes without problems and can be read by the WMG API? Perhaps in staging environment?



> @ebezzi @atarashansky
> 
> Currently, this would only verify that the reader can read data written by the `0.16.5` version of `tiledb`. We would also need to test:
> 
> 1. `tiledb 0.21.1` can be used by the WMG pipeline to generate cubes without problems
> 2. The WMG API can read the data generated in (1) above
> 
> Can we trigger the WMG pipeline run for this as well to ensure that the pipeline generates cubes without problems and can be read by the WMG API? Perhaps in staging environment?

I agree that we should also regenerate the cube using this version, but we'll need to get this merged before we do that since tiledb isn't forwards compatible.

File: requirements.txt [hunk 1]
```diff
@@ -22,4 +22,4 @@ s3fs==0.4.2
 SQLAlchemy-Utils>=0.36.8
 SQLAlchemy>=1.4.0,<1.5
 tenacity
-tiledb==0.16.5  # Portal's tiledb version should always be the same or older than Explorer's
+tiledb==0.21.1  # Portal's tiledb version should always be the same or older than Explorer's
```
Comment (ebezzi) [inline requirements.txt:25]:
```suggestion
tiledb==0.21.4  # Portal's tiledb version should always be the same or older than Explorer's
```

**Review by prathapsridharan — APPROVED** [review]
LGTM

File: requirements.txt [hunk 2]
```diff
@@ -22,4 +22,4 @@ s3fs==0.4.2
 SQLAlchemy-Utils>=0.36.8
 SQLAlchemy>=1.4.0,<1.5
 tenacity
-tiledb==0.21.1  # Portal's tiledb version should always be the same or older than Explorer's
+tiledb==0.21.4  # Portal's tiledb version should always be the same or older than Explorer's
```
[the remark this answers is not in the document]
Comment (prathapsridharan) [reply requirements.txt:25]: @atolopko-czi - @ebezzi just merged the explorer upgrade here: https://github.com/chanzuckerberg/single-cell-explorer/pull/617

**Pull request merged**